cowprotocol/cowswap
# feat(safe-eth-flow): safe eth flow wc banner

# Summary

Show WC Safe info banner when connected via WC to a Safe

Update text to include wrapping also for approval case

|Sell Native|Approval needed|
|-|-|
| ![Screenshot 2023-06-01 at 13 41 04](https://github.com/cowprotocol/cowswap/assets/43217/283293ee-4dd9-4182-87b2-a4b6025cc380) | ![Screenshot 2023-06-01 at 13 20 47](https://github.com/cowprotocol/cowswap/assets/43217/22b0d392-42d4-4af8-a9ea-3728ab362285) |

# To Test

1. Connect to a Safe via WC
2. Pick the native token as sell token and pick a buy token
* Aforementioned banner should be displayed
3. Pick as sell token one that needs approval
* Aforementioned banner should be displayed
4. Load the app as a Safe app
5. Repeat steps 1-3
* The bundle wrap banner should be displayed for selling eth
* The bundle approval banner should be displayed for selling a token that needs approval
6. Repeat steps 1-3 with an EOA/Different SC wallet
* No banners should be displayed
